report/3.Theory/SymbolTable.tex

```diff
@@ -0,0 +1,60 @@
+%THIS SECTION IS SUPPOSED TO BE A SUBSECTION TO THE PARSER SECTION.
+%Should present the interface, which all symbol table implementations must implement.
+%Should clarify all considerations through the design process.
+\subsection{Symbol Table}
+This section concerns with the implementation of the symbol table. 
+first, the interface implemented by the the symbol table will be presented.
+Second, all considerations through the design process will be discussed.
+This will serve as the reasoning behind the final implementation details.
+
+\textbf{Symbol Table Interface}\\
+The interface consists of five methods as seen in snippet 3.6 on page \pageref{lis:STInterface}.
+
+The methods on line 2 and 3 concerns with the handling of scopes.
+Scopes are handled by a stack data structure.
+Intuitively each element on the stack consists of a local symbol table concerning the current scope.
+
+$openScope()$ opens a new scope in the symbol table.
+This is done by pushing a new sub-symbol table onto the stack.
+
+$closeScope()$ closes the most recently opened sub-symbol table.
+This is done by popping the table from the stack.
+Symbol references subsequently revert to outer scopes.
+
+The methods on line 4 and 5 concerns with the handling of entering and retrieval of symbol information.
+Since retrieval is the most heavily utilized method in the symbol table implementation, it is essential that time spent on the procedure is minimal.
+Because of this all symbol tables uses a hash table data structure for insertion and retrieval of symbols. 
+
+$enterSymbol(name: String, type: TreeNode.Type): Boolean$ Enters a new symbol into the current scope on top of the stack.
```

Comment:
Either the Enters should be lower-case or add the "The entersymbol<blablabla> function enters....."

Reply:
fixed